test/e2e/simple_test.go

```diff
@@ -605,3 +605,47 @@ func TestSimpleBuildWithHybridSources(t *testing.T) {
 		t.Fatalf("Error watching build: %v", err)
 	}
 }
+
+// TestCancelledBuild tests that a build with a long running step
+// so that we have time to cancel it.
+func TestCancelledBuild(t *testing.T) {
+	logger := logging.GetContextLogger("TestCancelledBuild")
+
+	clients := buildClients(logger)
+	buildName := "cancelled-build"
+
+	test.CleanupOnInterrupt(func() { teardownBuild(clients, logger, buildName) }, logger)
+	defer teardownBuild(clients, logger, buildName)
+
+	if _, err := clients.buildClient.builds.Create(&v1alpha1.Build{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: buildTestNamespace,
+			Name:      buildName,
+		},
+		Spec: v1alpha1.BuildSpec{
+			Steps: []corev1.Container{{
+				Image: "busybox",
+				Args:  []string{"sleep", "5000"}, // fails.
+			}},
+		},
+	}); err != nil {
+		t.Fatalf("Error creating build: %v", err)
+	}
+
+	// Wait for a little while
+	// FIXME(vdemeester) I would prefer something less flaky-prone
+	time.Sleep(20 * time.Second)
```

Comment:
Probably consider adding polling function to check build status is "Running" for 10sec(? or more). That is better than wait solution IMO. WDYT @vdemeester ?

Reply:
@shashwathi yeah, that's why the FIXME :angel: not sure why I didn't use the polling functions here :sweat_smile: will update :wink: